chainerx_cc/examples/mnist_cc/train_mnist.cc

```diff
@@ -0,0 +1,212 @@
+    chx::Array train_x = ReadMnistImages(mnist_root + "train-images-idx3-ubyte");
+    chx::Array train_t = ReadMnistLabels(mnist_root + "train-labels-idx1-ubyte");
+    chx::Array test_x = ReadMnistImages(mnist_root + "t10k-images-idx3-ubyte");
+    chx::Array test_t = ReadMnistLabels(mnist_root + "t10k-labels-idx1-ubyte");
+
+    train_x = train_x.AsType(chx::Dtype::kFloat32) / 255.f;
+    train_t = train_t.AsType(chx::Dtype::kInt32);
+    test_x = test_x.AsType(chx::Dtype::kFloat32) / 255.f;
+    test_t = test_t.AsType(chx::Dtype::kInt32);
+
+    int64_t n_train = train_x.shape().front();
+    int64_t n_test = test_x.shape().front();
+
+    // Initialize the model with random parameters.
+    Model model{train_x.shape()[1], n_hidden, 10, n_layers};
+    std::random_device rd{};
+    std::mt19937 gen{rd()};
+    std::normal_distribution<float> dist{0.f, 0.05f};
+    model.Initialize(gen, dist);
+
+    auto start = std::chrono::high_resolution_clock::now();
+
+    for (int64_t epoch = 0; epoch < epochs; ++epoch) {
+        chx::Array train_indices = GetRandomIndicesArray(n_train, gen);
+
+        for (int64_t i = 0; i < n_train; i += batch_size) {
+            chx::Array indices = train_indices.At({chx::Slice{i, i + batch_size}});
+            chx::Array x = train_x.Take(indices, 0);
+            chx::Array t = train_t.Take(indices, 0);
+
+            chx::Backward(SoftmaxCrossEntropy(model(x), t));
+
+            // Vanilla SGD.
+            for (const chx::Array& param : model.params()) {
+                chx::Array p = param.AsGradStopped();
+                p -= param.GetGrad()->AsGradStopped() * lr;
+                param.ClearGrad();
+            }
+        }
+
+        // Evaluate.
+        {
+            chx::NoBackpropModeScope scope{};
+
+            chx::Array loss = chx::Zeros({}, chx::Dtype::kFloat32);
+            chx::Array acc = chx::Zeros({}, chx::Dtype::kFloat32);
+
+            for (int64_t i = 0; i < n_test; i += batch_size) {
+                std::vector<chx::ArrayIndex> indices{chx::Slice{i, i + batch_size}};
+                chx::Array x = test_x.At(indices);
+                chx::Array t = test_t.At(indices);
+                chx::Array y = model(x);
+                loss += SoftmaxCrossEntropy(y, t, false);
+                acc += (y.ArgMax(1).AsType(t.dtype()) == t).Sum().AsType(acc.dtype());
+            }
+
+            std::cout << "epoch: " << epoch << " loss=" << chx::AsScalar(loss / n_test) << " accuracy=" << chx::AsScalar(acc / n_test)
+                      << " elapsed_time=" << std::chrono::duration<double>{std::chrono::high_resolution_clock::now() - start}.count()
+                      << std::endl;
+        }
+    }
+}
+
+int main(int argc, char** argv) {
+    int64_t epochs{20};
+    int64_t batch_size{100};
+    int64_t n_hidden{1000};
+    int64_t n_layers{3};
+    float lr{0.01};
+    std::string device_name{"native"};
+    std::string mnist_root{"./"};
+
+    for (int i = 1; i < argc - 1; i += 2) {
```

Comment:
This is not robust enough because it does not generate any error if the value of each flag is omitted.